RecoHI/HiCentralityAlgos/plugins/HIhfFilter_miniAOD.cc

```diff
@@ -0,0 +1,115 @@
+
+#include <memory>
+
+#include "FWCore/Framework/interface/Frameworkfwd.h"
+#include "FWCore/Framework/interface/stream/EDProducer.h"
+
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
+
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/Utilities/interface/StreamID.h"
+
+#include "RecoLocalCalo/CaloTowersCreator/src/CaloTowerCandidateCreator.h"
+#include "DataFormats/CaloTowers/interface/CaloTowerDetId.h"
+#include "DataFormats/CaloTowers/interface/CaloTower.h"
+#include "DataFormats/CaloTowers/interface/CaloTowerCollection.h"
+#include "Geometry/Records/interface/CaloGeometryRecord.h"
+
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/ParameterSet/interface/ParameterSetDescription.h"
+#include "FWCore/ParameterSet/interface/ParameterSetDescriptionFiller.h"
+
+#include <vector>
+#include <cmath>
+#include "TMath.h"
+
+class HIhfFilter_miniAOD : public edm::stream::EDProducer<> {
+   public:
+      explicit HIhfFilter_miniAOD(const edm::ParameterSet&);
+      ~HIhfFilter_miniAOD() override;
+
+      static void fillDescriptions(edm::ConfigurationDescriptions& descriptions);
+
+   private:
+      virtual void beginStream(edm::StreamID) override;
+      virtual void produce(edm::Event&, const edm::EventSetup&) override;
+      virtual void endStream() override;
+
+      edm::EDGetTokenT<CaloTowerCollection>  srcTowers_;
+
+};
+
+HIhfFilter_miniAOD::HIhfFilter_miniAOD(const edm::ParameterSet& iConfig):
+  srcTowers_(consumes<CaloTowerCollection>(iConfig.getParameter<edm::InputTag>("srcTowers")))
+{
+   produces<std::vector<int>>("HIhfFilters");
+}
+
+
+HIhfFilter_miniAOD::~HIhfFilter_miniAOD()
+{
+}
+
+
+void
+HIhfFilter_miniAOD::produce(edm::Event& iEvent, const edm::EventSetup& iSetup)
+{
+  using namespace edm;
+  using namespace std; 
+  
+  Handle<CaloTowerCollection> towers;
+  iEvent.getByToken(srcTowers_, towers); 
+
+  int nTowersTh2HFplus_ =0;
+  int nTowersTh2HFminus_=0;
+  int nTowersTh3HFplus_ =0;
+  int nTowersTh3HFminus_=0;
+  int nTowersTh4HFplus_ =0;
+  int nTowersTh4HFminus_=0;
+  int nTowersTh5HFplus_ =0;
+  int nTowersTh5HFminus_=0;
+
+  std::size_t size = 4;
+  auto HIhfFiltersOut = std::make_unique<std::vector<int>>(size);
+  
+  for( unsigned int towerIndx = 0; towerIndx<towers->size(); ++towerIndx){
+    const CaloTower & tower = (*towers)[ towerIndx ];
+    if(tower.et() >= 0.0){
```

Comment:
How about:
```C++
const auto energy = tower.energy();
const auto eta = tower.eta();
const bool eta_plus = (eta>3.0) && (eta<6.0);
const bool eta_minus = (eta<-3.0) && (eta>-6.0);
if (eta_plus) {
  if (energy >= 2.0) nTowersTh2HFplus_ += 1;
  if (energy >= 3.0) nTowersTh3HFplus_ += 1;
...
} else if (eta_minus) {
  if (energy >= 2.0) nTowersTh2HFminus_ += 1;
  if (energy >= 3.0) nTowersTh3HFminus_ += 1;
}
```

Reply:
implemented in the commit b0ccac9877d896526f4a219906600508aa9cddef